pkg/cache/v2/cache.go

```diff
@@ -72,6 +87,23 @@ type Response interface {
 	GetVersion() (string, error)
 }
 
+// DeltaResponse is a wrapper around Envoy's DeltaDiscoveryResponse
+type DeltaResponse interface {
+	// Get the constructed DeltaDiscoveryResponse
+	GetDeltaDiscoveryResponse() (*discovery.DeltaDiscoveryResponse, error)
+
+	// Get the latest DeltaRequest for the DeltaResponse.
+	// Delta's requesting mechanism
```

Comment:
incomplete comment?

Reply:
Ah oops

Reply:
Does this make sense?
```go
	// Delta's requesting mechanism requires this to be the latest and not the original request
	// due to the protocol's knowledgement requirement of all past requests so the
	// management server can accurately create deltas on current resource maps
```

Reply:
How about

```
// The request that created the watch that we're now responding to. This is provided to allow the caller to correlate the
// response with a request. Generally this will be the latest request seen on the stream for the specific type.
```

I don't think we need to speak about the notion of original vs most recent, just speak to what it actually is and why it's provided

Reply:
Excellent I like that